src/instrumenter/state_var_instrumenter.ts

```diff
@@ -0,0 +1,1165 @@
+                    ["++", "--"].includes(updateNode.operator)),
+            `Only assignments and ++/-- return values. Not: ${pp(updateNode)}`
+        );
+
+        const value = factory.copy(stateVarExp);
+        const retStmt = factory.makeExpressionStatement(
+            factory.makeAssignment(
+                "<missing>",
+                "=",
+                factory.makeIdentifierFor(single(wrapperFun.vReturnParameters.vParameters)),
+                value
+            )
+        );
+        ctx.addGeneralInstrumentation(retStmt);
+
+        if (updateNode instanceof UnaryOperation && ["++", "--"].includes(updateNode.operator)) {
+            if (updateNode.prefix) {
+                body.appendChild(retStmt);
+            } else {
+                body.insertBefore(retStmt, body.children[0]);
+            }
+        } else {
+            assert(updateNode instanceof Assignment, ``);
+            body.appendChild(retStmt);
+        }
+    }
+
+    return wrapperFun;
+}
+
+/**
+ * Given the expression `e` make sure that `e` is contained in an `ExpressionStatement`, which itself
+ * is contained in a `Block`. There are several cases where we may need to create the block itself
+ * @param e
+ * @param factory
+ */
+function ensureToplevelExprInBlock(e: Expression, factory: ASTNodeFactory): void {
+    assert(e.parent instanceof ExpressionStatement, ``);
+    const container = e.parent.parent;
+    if (container instanceof Block || container instanceof UncheckedBlock) {
+        return;
+    }
+
+    if (container instanceof IfStatement) {
+        if (container.vTrueBody === e.parent) {
+            container.vTrueBody = factory.makeBlock([e.parent]);
+        } else {
+            assert(container.vFalseBody === e.parent, ``);
+            container.vFalseBody = factory.makeBlock([e.parent]);
+        }
+        return;
+    }
+
+    if (container instanceof ForStatement) {
+        assert(
+            container.vBody === e.parent,
+            `Currently dont support instrumenting tuple assignments in for init/loop expession`
+        );
+        container.vBody = factory.makeBlock([e.parent]);
+        return;
+    }
+
+    if (container instanceof WhileStatement) {
+        assert(
+            container.vBody === e.parent,
+            `Currently dont support instrumenting tuple assignments in for init/loop expession`
+        );
+        container.vBody = factory.makeBlock([e.parent]);
+        return;
+    }
+
+    if (container instanceof DoWhileStatement) {
+        assert(
+            container.vBody === e.parent,
+            `Currently dont support instrumenting tuple assignments in for init/loop expession`
+        );
+        container.vBody = factory.makeBlock([e.parent]);
+        return;
+    }
+}
+
+/**
+ * Complex interposing case - when `updateNode` is a (potentially nested) tuple assignment with multiple
+ * state vars and other expressions updated toghether.
```

Comment:
typo `toghether -> together`, or is there some ethereum thing called toghether(togh-ether) as I have seen the same in `walk.ts` and `scribble_cli.json`

Reply:
Fixed in 35d34c0.